src/rp/index.ts

```diff
@@ -0,0 +1,143 @@
+import { AuthRequestDecodeError, IoWalletError } from "../utils/errors";
+import {
+  decode as decodeJwt,
+  decodeBase64,
+  sha256ToBase64,
+  SignJWT,
+} from "@pagopa/io-react-native-jwt";
+import { RequestObject, RpEntityConfiguration } from "./types";
+
+import uuid from "react-native-uuid";
+import type { JWK } from "@pagopa/io-react-native-jwt/lib/typescript/types";
+
+export class RelyingPartySolution {
+  relyingPartyBaseUrl: string;
+  walletInstanceAttestation: string;
+  appFetch: GlobalFetch;
+
+  constructor(
+    relyingPartyBaseUrl: string,
+    walletInstanceAttestation: string,
+    appFetch: GlobalFetch = { fetch }
+  ) {
+    this.relyingPartyBaseUrl = relyingPartyBaseUrl;
+    this.walletInstanceAttestation = walletInstanceAttestation;
+    this.appFetch = appFetch;
+  }
+
+  /**
+   * Decode a QR code content to an authentication request url.
+   * @function
+   * @param qrcode QR code content
+   *
+   * @returns The authentication request url
+   *
+   */
+  decodeAuthRequestQR(qrcode: string): string {
+    let decoded = decodeBase64(qrcode);
+    let decodedUrl = new URL(decoded);
+    let requestUri = decodedUrl.searchParams.get("request_uri");
+    if (requestUri) {
+      return requestUri;
+    } else {
+      throw new AuthRequestDecodeError(
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Multiple steps of this method can fail. Should we wrap the whole body in a try/catch and raise a consistent error type?